web/src/stream_popover.js

```diff
@@ -360,24 +383,36 @@ export function build_move_topic_to_stream_popover(
         $stream_header_colorblock = $("#dialog_widget_modal .topic_stream_edit_header").find(
             ".stream_header_colorblock",
         );
-        stream_bar.decorate(current_stream_name, $stream_header_colorblock);
-        const streams_list =
-            message_edit.get_available_streams_for_moving_messages(current_stream_id);
-        const opts = {
-            widget_name: "select_stream",
-            data: streams_list,
-            default_text: $t({defaultMessage: "No streams"}),
-            include_current_item: false,
-            value: current_stream_id,
-            on_update: move_topic_on_update,
-        };
-        stream_widget = new DropdownListWidget(opts);
-
-        stream_widget.setup();
+        stream_widget_value = current_stream_id;
+        const streams_list_options = () =>
+            stream_data.get_options_for_dropdown_widget().filter((stream) => {
+                if (stream.stream_id === current_stream_id) {
+                    return true;
+                }
+                return stream_data.can_post_messages_in_stream(stream);
+            });
+        dropdown_widget.setup(
+            {
+                target: "#move_topic_stream_widget",
+                // Overlap dropdown search input with stream selection button.
+                placement: "bottom-start",
+                offset: [0, -30],
+            },
+            streams_list_options,
+            move_topic_on_update,
+        );
```

Comment:
OK a few notes on the API here that I think we should improve:
* Let's try to rework the `dropdown_widget` API to used named parameters -- `move_topic_on_update` should ideally be an `on_update` parameter like we had with `DropdownListWidget`
* I think it'd probably be nice to make `dropdown_widget` a class component defined as a class like `DropdownListWidget` was; that API feels a lot cleaner here.
* I think we may want to make a `StreamPickerDropdownWidget` and later `GroupPickerDropdownWidget` subclass component that basically contains the `render_selected_stream` part of the implementation and includes a CSS class that allows us to have most of the CSS you have below apply to the reusable component that we're going to have in settings as well; with the possible exception of the compose box one, I think we intend for all of our stream pickers to be the same, and it'd be nice to have the code express that.

Reply:
Sounds good.